evennia/plugin.py

```diff
@@ -0,0 +1,217 @@
+        """
+        Runs in the reverse order of load_priority, but before the 'mygame' hooks.
+
+        see at_server_startstop module documentation.
+        """
+
+    def at_server_cold_start(self):
+        """
+       Runs in the same order as load_priority, but before the 'mygame' hooks.
+
+       see at_server_startstop module documentation.
+       """
+
+    def at_server_cold_stop(self):
+        """
+        Runs in the reverse order of load_priority, but before the 'mygame' hooks.
+
+        see at_server_startstop module documentation.
+        """
+
+
+class PluginManager:
+    """
+    Basic class used for maintaining an index of Plugins installed.
+    """
+
+    def __init__(self):
+        self.plugins = dict()
+        self.plugins_sorted = list()
+        self.reverse = None
+
+    def add(self, plugin):
+        self.plugins[plugin.name()] = plugin
+        self.plugins_sorted.append(plugin)
+
+    def sort(self):
+        self.plugins_sorted.sort(key=lambda x: x.load_priority())
+        self.reverse = reversed(self.plugins_sorted)
+
+    def __getitem__(self, item):
+        return self.plugins[item]
+
+    def init_settings(self, settings):
+        for plugin in self.plugins_sorted:
+            plugin.init_settings(settings, self.plugins)
+
+    def at_plugin_setup(self):
+        for plugin in self.plugins_sorted:
+            plugin.at_plugin_setup()
+
+    def at_plugin_collaborate(self):
+        for plugin in self.plugins_sorted:
+            plugin.at_plugin_collaborate()
+
+    def at_plugin_finalize(self):
+        for plugin in self.plugins_sorted:
+            plugin.at_plugin_finalize()
+
+    def at_server_start(self):
+        for plugin in self.plugins_sorted:
+            plugin.at_server_start()
+
+    def at_server_stop(self):
+        for plugin in self.reverse:
+            plugin.at_server_stop()
+
+    def at_server_reload_start(self):
+        for plugin in self.plugins_sorted:
+            plugin.at_server_reload_start()
+
+    def at_server_reload_stop(self):
+        for plugin in self.reverse:
+            plugin.at_server_reload_stop()
+
+    def at_server_cold_start(self):
+        for plugin in self.plugins_sorted:
+            plugin.at_server_cold_start()
+
+    def at_server_cold_stop(self):
+        for plugin in self.reverse:
+            plugin.at_server_cold_stop()
+
+
+def import_property(path):
+    """
+    Stripped-down version of evennia.utils.utils.class_from_module that gets along with
+    django.conf.settings without triggering its setup prematurely.
+
+        Hopefully, a function object or a variable or something of that sort.
+    """
+    import importlib
+    if '.' in path:
+        module, thing = path.rsplit('.', 1)
+        module = importlib.import_module(module)
+        thing = getattr(module, thing)
+        return thing
+    else:
+        return importlib.import_module(path)
+
+
+_already_configured = False
+
+
+def plugin_settings(settings_path):
```

Comment:
This name is not very useful. Better call it `init_plugin_settings` because that's what it does.